docs/README.md

```diff
@@ -0,0 +1,56 @@
+## Sequence Diagram
+
+This is how the user interacts with the various components of this application.
+
+The different components have been chosen with specific needs in mind:
+* Backend: The backend is needed to protect the private key of the wallet that is doing the actual signing of the transactions on the blockchain.
+* Forwarder: Ensures that the incoming transaction gets validated before actually sent for execution. The forwarder also keeps a record of the internal nonces that external wallets are using while forwarding.
+* Recipient: This is the final smart contract whose method is originally called from the Frontend. It substitutes the forwarder sender for the actual sender (from) of the original transaction and persists the data on-chain.
+
+```mermaid
+  sequenceDiagram
+    participant Frontend
+    participant Metamask
+    participant User
+    participant Backend
+    participant Forwarder
+    participant Recipient
+
+    rect rgb(111, 160, 200, 0.8)
+        Frontend-->>Recipient: Subscribe to Interaction Events
+    end
+
+    activate Metamask
+    rect rgb(208, 240, 192)
+      Frontend->>Metamask: Request Accounts
+      Metamask->>User: Ask to select wallet <br /> and connect
+      User->>Frontend: selects wallet and connects
+    end
+    rect rgb(208, 240, 192)
+      Frontend->>Metamask: Request to INTERACT
```

Comment:
Maybe add another interaction after Request to Interact between User->>Frontend: Inputs a message and press Send button

Reply:
Try not to mistake the User for an Actor; the "User" box describes the User's interaction with Metamask. The User's interaction with the Frontend is depicted by the "Frontend" box.

Reply:
okay i will keep that in mind.